xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -294,8 +303,14 @@ public String getQuery(XWikiContext context)
                         select.append(", doc." + valueField);
                     } else {
                         select.append(", valueprop.value");
-                        fromStatements.add("StringProperty as valueprop");
                         whereStatements.add("obj.id=valueprop.id.id and valueprop.id.name='" + valueField + "'");
+                        // Get the from statements according to the type of property
+                        try {
+                            String classType = getPropertyType(classname, valueField,context);
+                            fromStatements.add(classType + " as valueprop");
+                        } catch (Exception e) {
+                            LOGGER.error("Failed to parse SQL script [{}]. Continuing with non-rendered script.", sql, e);
```

Comment:
Same as above, the log message doesn't seem related to the code executed in the ``try`` block above and I don't see what exceptions are you hoping to catch here.